### PR TITLE
feat(oidc): add oidc verification method v2

### PR DESCRIPTION
**Pull request by JMounier**

### CODE REVIEW

File: protocol.md [hunk 1]
```diff
@@ -281,6 +281,24 @@ The steps to verify the *user* using OpenID Connect are as follows:
 
 The process to [register a new device](#device-registration) with an `OIDC` [Verification Method] is the same as described above. The only difference is that at the end of the process the *Tanker server* returns the *user*'s [Verification Key].
 
+### OpenID Connect with authorization code flow
+
+The *Tanker Server* is integrated with a list of trusted OpenID Connect provider. The *Tanker Server* uses its own `client ID` and `client secret` to act as a `service provider`. It follows the [the OpenID protocol](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth).
```
Comment (TimotheeIsnard) [inline protocol.md]: provider_s_

follows the the

File: protocol.md [hunk 2]
```diff
@@ -281,6 +281,24 @@ The steps to verify the *user* using OpenID Connect are as follows:
...
+
+For the *application* to offer OpenID Connect as a verification method, *application* owners need to register the trusted OpenID Connect provider to the *Tanker App*.
```
Comment (TimotheeIsnard) [inline protocol.md]: register to -> register with/register in, I think?

File: protocol.md [hunk 3]
```diff
@@ -281,6 +281,24 @@ The steps to verify the *user* using OpenID Connect are as follows:
...
+
+The steps to verify the *user* using OpenID Connect are as follows:
+1. The *user* queries the `OIDC` sign-in endpoint of the *Tanker server*
+1. The *Tanker server* prepares an authentication request and redirects the user to the OpenID Connect provider
+1. OpenID Connect provider authenticates the *user* and obtains their consent/authorization
+1. OpenID Connect provider responds with an `authorization code`
```
Comment (TimotheeIsnard) [inline protocol.md]: The OpenID connect provider

File: protocol.md [hunk 4]
```diff
@@ -281,6 +281,24 @@ The steps to verify the *user* using OpenID Connect are as follows:
...
+1. *Tanker Core* sends a request to the *Tanker server*. The request contains the `authorization code`
+1. The *Tanker server* requests, from the OpenID Connect provider, an `ID Token` in exchange for the `authorization code` and *Tanker server*'s `client ID` and `client secret`
```
Comment (TimotheeIsnard) [inline protocol.md]: and the Tanker server's

File: protocol.md (hunk 2 above)
Comment (TimotheeIsnard) [reply protocol.md]: (also Verification Method is capitalized and linkified everywhere else, maybe we also want to do it here?)

**Review by TimotheeIsnard — APPROVED** [review]

**Review by tux3 — APPROVED** [review]

**Pull request merged**